x-pack/plugins/uptime/server/lib/synthetics_service/get_api_key.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getAPIKeyForSyntheticsService } from './get_api_key';
+import { encryptedSavedObjectsMock } from '../../../../encrypted_saved_objects/server/mocks';
+import { securityMock } from '../../../../security/server/mocks';
+import { coreMock } from '../../../../../../src/core/server/mocks';
+import { syntheticsServiceApiKey } from '../saved_objects/service_api_key';
+import { KibanaRequest } from 'kibana/server';
+
+describe('getAPIKeyTest', function () {
+  const core = coreMock.createStart();
+  const security = securityMock.createStart();
+  const encryptedSavedObject = encryptedSavedObjectsMock.createStart();
+  const request = {} as KibanaRequest;
+
+  security.authc.apiKeys.areAPIKeysEnabled = jest.fn().mockReturnValue(true);
+  security.authc.apiKeys.create = jest.fn().mockReturnValue({
+    id: 'test',
+    name: 'service-api-key',
+    api_key: 'qwerty',
+    encoded: '@#$%^&',
+  });
+
+  it('should generate an api key and return it', async () => {
+    const apiKey = await getAPIKeyForSyntheticsService({
+      request,
+      security,
+      encryptedSavedObject,
+      savedObjectsClient: core.savedObjects.getScopedClient(request),
+    });
+
+    expect(security.authc.apiKeys.areAPIKeysEnabled).toHaveBeenCalledTimes(1);
+    expect(security.authc.apiKeys.create).toHaveBeenCalledTimes(1);
+    expect(security.authc.apiKeys.create).toHaveBeenCalledWith(
+      {},
+      {
+        name: 'synthetics-api-key',
+        role_descriptors: {
+          synthetics_writer: {
+            cluster: ['monitor', 'read_ilm', 'read_pipeline'],
+            index: [
+              {
+                names: ['synthetics-*', 'heartbeat-*'],
```

Comment:
Should we be indexing to both `heartbeat-*` and `synthetics-*`? I was under the impression we should only be indexing to `synthetics-*`

Reply:
I'm realizing that I'm commenting this on the test, but it applies to the actual implementation too. Same for the below comment.

Reply:
since we are going to use data stream format, it makes sense to remove heartbeat, i will do that

Reply:
Done